src/lib/optimizer/strategy/between_composition_rule.cpp

```diff
@@ -348,12 +348,12 @@ void BetweenCompositionRule::_replace_predicates(const std::vector<std::shared_p
   }
 }
 
-void BetweenCompositionRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
-  if (node->type == LQPNodeType::Predicate) {
+void BetweenCompositionRule::_apply_to(const std::shared_ptr<AbstractLQPNode>& lqp_root) const {
```

Comment:
Haven't looked at the individual changes to the rules yet, as those are subject to how we name the interfaces.